### DISCUSSION TITLE
How to filter a blog by tags/taxonomies dynamically (frontend)

### DISCUSSION THREAD

**Discussion by pathly-dev** [statamic/cms#5243] [Q&A] [answered]
Hi, I have a blog (collection) with a lot of blog posts (entries) and each entry has different tags (taxonomy terms) – like color or size – so you can filter by specific tags via a select in the frontend, e.g. I want to filter by the color 'red' and the size 'big'. 
Now I'm still pretty new to Statamic and I don't know how to do that.

I know that you can filter like 
```
{{ collection:blog color:is="red" }}
   ...
{{ /collection:blog }}
```

But I don't want to do that hard coded, it should be dynamically filterable by the user.

Some ideas I found were using the API or GET, but I don't understand how and how to use, since the docs aren't that newby-friendly.

Hopefully someone can help :)

**Comment by jackmcdade** [comment, marked as answer]
If you want to use `GET` params in the URL (my preference), you can do something like this:

```
// example.com/blog?color=blue&size=massive
{{ collection:blog :taxonomy:color="get:color" :taxonomy:size="get:size" }}

{{ /collection:blog }}
```

**Reply by pathly-dev** [reply]
@jackmcdade Ok thanks, that is what i found so far. But how do these parameters come into the url? That is something I don’t understand and didn‘t found.

**Reply by jackmcdade** [reply]
That's custom JavaScript land – not something Statamic would do natively. You would basically wire up the select input with change events that push the value to query strings in the URL.

**Reply by pathly-dev** [reply]
Ah ok, I understand! Thanks for your help! :)

**Reply by pathly-dev** [reply]
I tried this out, but it doenst work. When I try this instead
```
{{ collection:blog :color:is="get:color" }}
{{ /collection:blog }}
```
 it is working partially. But then it doenst show anything when there is no filter set, only when I selected one.

I'm not sure, what I am doing wrong. The taxonomy is called color, also the field of the entry.

**Comment by jackmcdade** [comment]
Wrap the tag with a check for get:color and that should do the trick.
